controllers/addresspool_controller.go

```diff
@@ -110,45 +113,23 @@ func (r *AddressPoolReconciler) syncMetalLBAddressPool(instance *metallbv1alpha1
 	return err
 }
 
-func (r *AddressPoolReconciler) syncMetalLBAddressPools(req ctrl.Request) error {
+func (r *AddressPoolReconciler) syncMetalLBAddressPools() error {
 	instanceList := &metallbv1alpha1.AddressPoolList{}
-	objs := make([]*unstructured.Unstructured, 0)
-
-	configMap := &corev1.ConfigMap{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "config",
-			Namespace: req.Namespace,
-		},
-	}
-
-	// Delete the exiting configMap
-	if err := r.Delete(context.Background(), configMap); err != nil {
-		// if we don't have ConfigMap then there is nothing to do
-		if errors.IsNotFound(err) {
-			return nil
-		}
-		r.Log.Info(fmt.Sprintf("Failed to delete existing Configmap %s", err))
-		return err
-	}
+	specList := apply.ConfigMapData{}
```

Comment:
this can be an []AddressPoolSpec I think, same for the peers

Reply:
yeah I tried both yesterday I left this way to be consistent with the modifier but both worked curious why u prefer the above ?